### PR TITLE
Add execute_sql function to enable execution of SQL code where no DataFrame is returned

### PR DESCRIPTION
**Pull request by madhav-datt**
Initial pull request based on the discussion and issue explained [here](https://github.com/pydata/pandas-gbq/issues/45). I'd love to discuss any changes to the design of this function, or if it might be better to add this functionality elsewhere.

### CODE REVIEW

File: pandas_gbq/gbq.py [hunk 1]
```diff
@@ -874,6 +874,94 @@ def _parse_entry(field_value, field_type):
...
+
+    """
+
+    try:
+        read_gbq(query, project_id=project_id, index_col=index_col,
+                 col_order=col_order, reauth=reauth, verbose=verbose,
+                 private_key=private_key,
+                 auth_local_webserver=auth_local_webserver,
+                 dialect=dialect, **kwargs)
+    except KeyError:
```
Comment (MaximilianR) [inline pandas_gbq/gbq.py]: I think it's mostly coincidental that the existing function raises a `KeyError`. Can we find a more coherent way of doing this?

**Comment by tswast** [comment]
Could you rebase this PR on the latest version in master? There have been some significant changes since this was opened. We changed what library we use to actually make the API calls to BigQuery.

**Comment by madhav-datt** [comment]
@tswast Thanks so much for the update. I'll change this PR based on the latest code in master.

**Comment by maxim-lian** [comment]
@madhav-datt Would love to merge this. Let us know if you're up for updating this.

**Comment by tswast** [comment]
I've been thinking about this in regards to `dryRun` queries, too. (https://github.com/pydata/pandas-gbq/issues/88) I think it might make some sense to return an empty dataframe in the case when there are no result rows.

**Comment by madhav-datt** [comment]
Thanks for the follow up! Sure - I'll go ahead and update the pull request.

**Comment by max-sixty** [comment]
@madhav-datt this is v close! How would you feel about pushing over the line?

**Comment by tswast** [comment]
Thank you for the contribution @madhav-datt and your patience. I'm closing this in favor of https://github.com/pydata/pandas-gbq/pull/286 which adds a `max_results` argument to `read_gbq`. Set `max_results=0` if no results are desired.

**Pull request closed without merging**